db/repos/blocks.js

```diff
@@ -23,110 +25,238 @@ var columnSet;
  * @param {Database} db - Instance of database object from pg-promise
  * @param {Object} pgp - pg-promise instance to utilize helpers
  * @constructor
- * @return {BlocksRepo}
+ * @return {BlocksRepository}
  */
-function BlocksRepo (db, pgp) {
-	this.db = db;
-	this.pgp = pgp;
-
-	this.dbTable = 'blocks';
-
-	this.sortFields = [
-		'id',
-		'timestamp',
-		'height',
-		'previousBlock',
-		'totalAmount',
-		'totalFee',
-		'reward',
-		'numberOfTransactions',
-		'generatorPublicKey'
-	];
-
-	this.dbFields = [
-		'id',
-		'version',
-		'timestamp',
-		'height',
-		'previousBlock',
-		'numberOfTransactions',
-		'totalAmount',
-		'totalFee',
-		'reward',
-		'payloadLength',
-		'payloadHash',
-		'generatorPublicKey',
-		'blockSignature'
-	];
-
-	if (!columnSet) {
-		columnSet = {};
-		var table = new pgp.helpers.TableName({table: this.dbTable, schema: 'public'});
-		columnSet.insert = new pgp.helpers.ColumnSet(this.dbFields, {table: table});
+class BlocksRepository {
+
+	constructor (db, pgp) {
+
+		this.db = db;
+		this.pgp = pgp;
+
+		if (!cs.insert) {
+
+			const columns = [
+				'id',
+				'version',
+				'timestamp',
+				'height',
+				'previousBlock',
+				'numberOfTransactions',
+				'totalAmount',
+				'totalFee',
+				'reward',
+				'payloadLength',
+				'payloadHash',
+				'generatorPublicKey',
+				'blockSignature'
+			];
+
+			const table = new pgp.helpers.TableName({table: 'blocks', schema});
+			cs.insert = new pgp.helpers.ColumnSet(columns, {table});
+		}
+	}
+
+	// TODO: Merge BlocksRepository#getGenesisBlock with BlocksRepository#getGenesisBlockId
+	/**
+	 * Get the genesis block
+	 * @return {Promise}
+	 */
+	getGenesisBlock () {
+		return this.db.any(sql.getGenesisBlock);
+	}
+
+	/**
+	 * Get genesis block by id
+	 * @param {string} id
+	 * @return {Promise}
+	 */
+	getGenesisBlockId (id) {
+		// TODO: Must use a result-specific method, not .query
+		return this.db.query(sql.getGenesisBlockId, [id]);
```

Comment:
Yes, that'd be my guess also, but this is not the change to be done within this PR, because we will need also update the code that's using this method, which is outside the scope here.

Reply:
Understandable.